### PR TITLE
Add the ability to run syft from a scratch image.

### PR DESCRIPTION
**Pull request by Toure**
This change will allow endusers or CI to run syft from a
minimum image which will simplify CI deployment.

closes: #228 
Signed-off-by: Toure Dunnon <toure.dunnon@anchore.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN apk --no-cache add ca-certificates
+
+# create empty directory for scratch image cache.
+RUN mkdir -p /tmp-syft
+
+# reduce container image to scratch size.
+FROM scratch
+
+# Copy directories and files needed to execute syft.
+COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```
Comment (luhring) [inline Dockerfile:13]: What do we use these certs for?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+COPY --from=build /tmp-syft /tmp
```
Comment (luhring) [inline Dockerfile:14]: One option here would be to create the `/tmp` directory using `WORKDIR`. There's an argument to be made that this is a bit of a hack. It'd also simplify the Dockerfile and possibly remove the need for the multi-stage approach altogether. Example:

```Dockerfile
FROM scratch

WORKDIR /tmp  # used only to create this directory
WORKDIR /

COPY ./syft /

ENTRYPOINT [ "/syft" ]
```

**Review by luhring — COMMENTED** [review]
Nice work! I'd also suggest that adding documentation for how to use this Docker image is a "must have" for this addition, since it may not be obvious how to use it (e.g. that most workflows would require access to the Docker daemon).

File: Dockerfile (hunk 1 above)
Comment (wagoodman) [reply Dockerfile:13]: when syft checks for updates it does so over https, these certs are needed for verifying this trust.

**Comment by wagoodman** [comment]
ditto on @luhring s comment for documentation, just a small example or two in the readme will go a long way.

File: Dockerfile (hunk 2 above)
Comment (wagoodman) [reply Dockerfile:14]: IMHO since the multi-stage approach is still required the `COPY` approach is OK (and is generally more explicit)

Comment (luhring) [reply Dockerfile:14]: Agreed

File: README.md [hunk 3]
```diff
@@ -69,6 +69,16 @@ brew tap anchore/syft
 brew install syft
 ```
 
+**Container Image**
+
+Example of how to use Syft scratch image:
```
Comment (wagoodman) [inline README.md:74]:
```suggestion
Example of how to use Syft with docker:
```

File: README.md [hunk 4]
```diff
@@ -69,6 +69,16 @@ brew tap anchore/syft
...
+
+
+```bash
+docker pull anchore/syft:latest
```
Comment (wagoodman) [inline README.md:78]:
```suggestion

```
technically docker will attempt to pull the image, so saves example space

File: README.md [hunk 5]
```diff
@@ -69,6 +69,16 @@ brew tap anchore/syft
...
+docker run -v /var/run/docker.sock:/var/run/docker.sock anchore/syft -o json alpine:latest
```
Comment (wagoodman) [inline README.md:79]:
```suggestion
docker run -v /var/run/docker.sock:/var/run/docker.sock anchore/syft:latest -o json alpine:latest
```

Comment (wagoodman) [reply README.md:79]: also, should we throw in `-t` to get the ETUI output? or leave as is for logger output? (open question)

File: README.md [hunk 6]
```diff
@@ -69,6 +69,16 @@ brew tap anchore/syft
 brew install syft
 ```
 
+**Container Image**
```
Comment (wagoodman) [inline README.md:72]:
```suggestion
**Docker**
```

File: README.md (hunk 3 above)
Comment (wagoodman) [reply README.md:74]: (in this way we just use the bolded title/section to hint at the installation mode)

File: README.md (hunk 6 above)
Comment (alfredodeza) [reply README.md:72]: Why do we want to say Docker instead of container image?

File: README.md (hunk 5 above)
Comment (Toure) [reply README.md:79]: So I added the "-t" or Allocate a pseudo TTY option and the terminal hung, which is odd.

File: README.md (hunk 6 above)
Comment (Toure) [reply README.md:72]: I think container image is a better term as this will allow us to include other engines in the future.

Comment (wagoodman) [reply README.md:72]: actually, on further consideration, I retract this...

File: README.md (hunk 5 above)
Comment (wagoodman) [reply README.md]: gotcha, we'll leave this as is 👍  thanks for confirming

**Review by wagoodman — APPROVED** [review]
looks great!

**Pull request merged**